torch/csrc/jit/register_special_ops.cpp

```diff
@@ -27,6 +27,30 @@ RegisterOperators reg({
         [](Stack& stack) {
           return 0;
         }),
+    Operator(
+        "aten::size(Tensor self) -> int[]",
+        [](Stack& stack) {
+          autograd::profiler::RecordFunction record("sizes");
+          auto result = (std::move(pop(stack))).toTensor().sizes();
+          pack(stack, std::move(result));
+          return 0;
+        }),
+    Operator(
+        "aten::list_with_default(int[] list, int[] defaults) -> int[]",
```

Comment:
I think this should be expressed in torchscript

Reply:
`        @torch.jit.script
        def _list_with_default(out_size, defaults):
            # type: (List[Optional[int]], List[int]) -> List[int]

            if len(defaults) <= len(out_size):
                raise ValueError('Input dimension should be at least {}'.format(len(out_size) + 1))

            out = torch.jit._construct_empty_int_list()
            for i in range(len(out_size)):
                out_i = out_size[i]
                if out is not None:
                    out.append(torch.jit._unwrap_optional(out_i))
                else:
                    out.append(defaults[i])
            return out
`

Reply:
There's this in `_list_with_default`

```python
def _list_with_default(out_size, defaults):
    if isinstance(out_size, int):
        return out_size
```

So the semantics are slightly changed but I think it doesn't actually matter since a `int` return gets converted to a list before the actual op is called

Reply:
Yea agreed it should be fine.  The equivalent condition (since the int gets broadcasted) is:

`if len(out_size) == 1 return out_size` at the beginning of the function.

Reply:
It looks like in the actual usages of it, out_size never has None values.